lib/spack/spack/installer.py

```diff
@@ -1791,8 +1791,12 @@ def build_process(pkg, kwargs):
         # Run post install hooks before build stage is removed.
         spack.hooks.post_install(pkg.spec)
 
-    # Stop the timer
-    pkg._total_time = time.time() - start_time
+    # Stop the timer and save results
+    timer.stop()
+    with open(pkg.times_log_path, 'w') as timelog:
+        timer.write_json(timelog)
+
+    pkg._total_time = timer.total
```

Comment:
Pretty sure we do not need these variables anymore -- they can be removed from `PackageBase.__init__` and here.  I was expecting the `tty.msg()` call below to just end up being a call to `Timer.write_tty()`.

Reply:
The message for the installer and the solver are different, I'm not sure it makes sense to shove them both into the same function. I would have to make the message much more generic, which I think is not as meaningful/useful for the reader.